src/map/mob.cpp

```diff
@@ -2809,6 +2809,41 @@ int32 mob_getdroprate(struct block_list *src, std::shared_ptr<s_mob_db> mob, int
 	return drop_rate;
 }
 
+/**
+ * Returns the MVP player based on the monster's damage log
+ * This player has the highest value when damage dealt and damage tanked are added together
+ * @return The MVP player
+ */
+map_session_data* mob_data::get_mvp() {
+	// There cannot be an MVP player if the monster is not an MVP
+	if (this->get_bosstype() != BOSSTYPE_MVP)
+		return nullptr;
+
+	int64 mvp_damage = 0;
+	map_session_data* mvp_sd = nullptr;
+
+	for (int32 i = 0; i < this->dmglog.size(); i++) {
+		const s_dmglog& entry = this->dmglog[i];
+
+		map_session_data* tsd = map_charid2sd(entry.id);
+
+		if (tsd == nullptr)
+			continue; // skip players that are offline
+		if (tsd->bl.m != this->bl.m)
+			continue; // skip players not on this map
+		if (pc_isdead(tsd))
+			continue; // skip dead players
+
+		if (mvp_damage >= entry.dmg + entry.dmg_tanked)
+			continue;
+
+		mvp_damage = entry.dmg + entry.dmg_tanked;
```

Comment:
Shouldn't we use `util::safe_addition_cap` ?

Reply:
True

Reply:
Unlikely that these will exceed int64. Also Aegis would also just loop into the negatives if that happened, unless they changed the code when they moved from int32 to another type.

Reply:
Does not matter. Please use it.